tests/cephadm/test_cephadm_upgrade.py

```diff
@@ -80,6 +82,17 @@ def run(ceph_cluster, **kwargs) -> int:
         # Monitor upgrade status, till completion
         orch.monitor_upgrade_status()
 
+        if config.get("verify_cephadm_containers") and is_legacy_container_present(
+            ceph_cluster
+        ):
+            log.info(
+                "Checking cluster status to ensure that the legacy services are not being inferred"
+            )
+            if orch.cluster.check_health(
+                rhbuild=config.get("rhbuild"), client=orch.installer
+            ):
+                raise UpgradeFailure("Cluster is in HEALTH_ERR state after upgrade")
```

Comment:
Are you sure the cluster turn to be `HEALTH_ERR` state if this test fails?

Reply:
Yes.